Stewart et al. Supplementary Code.Rmd

```diff
@@ -1,243 +1,252 @@
-coll <- subset(df5, coll=="TRUE")
-
-coll <- coll %>%
-  group_by(Categories) %>%
-  mutate(Categories = str_replace(Categories, ".*?\\s", "")) %>%
-  mutate(count=n())
-
-coll_dtm <- coll %>%
-  cast_dtm(DOI.name, Categories, count)
-
-w <- dcast(coll, Categories~DOI.name)
-x <- as.matrix(w[,-1])
-x[is.na(x)] <- 0
-x <- apply(x, 2,  function(x) as.numeric(x > 0))
-coocCounts <- x %*% t(x) ### create matrix
-dimnames(coocCounts) <- list(w[, 1], w[,1])
-coocCounts <- as.data.frame(coocCounts)
-
-source("https://slcladal.github.io/rscripts/calculateCoocStatistics.R")
-
-coocTerm <- "Impact"
-coocs <- calculateCoocStatistics(coocTerm, coll_dtm, measure="DICE")
-
-coocdf <- coocs %>%
-  as.data.frame() %>%
-  dplyr::mutate(CollStrength = coocs,
-                Term = names(coocs))
-```
-
-## Collectors curves
-The below example produces a collectors curve for concepts. To select other variable use the following in place of "^4.":
-
-Variable | Search String |
---- | --- |
-Data type | ^5. |
-Theme | ^7. |
-Event type | ^12. |
-
-```{r, collectors, message=FALSE, warning=FALSE}
-df6 <- df
-df6 <- separate_rows(df6, Categories, sep = ";")
-df6$Categories <- trimws(df6$Categories, which = c("left"))
-df6$concept <- str_detect(df6$Categories, "^4.")
-concept <- subset(df6, concept=="TRUE")
-
-concept <- as.data.frame(with(concept, unclass(table(DOI.name, Categories))))
-
-SAC <- specaccum(concept, method = "random", permutations = 100)
-plot(SAC, xlab = "Articles", ylab = "Concepts", col="#045a8d", lwd=1, ci.col="#9ecae1")
-```
-
-## Cluster analysis
-Performs a cluster analysis for the most common themes and produces a heat map.
-
-```{r, cluster, message=FALSE, warning=FALSE}
-df7 <- read.csv("Appendix_4.csv")
-df7 <- separate_rows(df7, Categories, sep = ";")
-df7$Categories <- trimws(df7$Categories, which = c("left"))
-
-df7$theme <- str_detect(df7$Categories, "^7.")
-theme <- subset(df7, theme=="TRUE")
-theme <- theme %>%
-  group_by(Categories) %>%
-  mutate(count=n()) %>%
-  mutate(Categories = str_replace(Categories, ".*?\\s", ""))
-
-theme <- subset(theme, count > 18)
-theme <- as.data.frame(with(theme, unclass(table(Author, Categories))))
-heatmap.2(as.matrix(theme), key=FALSE, trace = "none", col = c("white", "black"), margins = c(11,11))
-```
+---
+title: "Replication document for Extreme Events Review"
+author: "Mathew Stewart"
+output: pdf_document
+---
+
+## Load Libraries
+
+```{r libraries, message=FALSE}
+knitr::opts_chunk$set(
+  message = FALSE,
+  warning = FALSE
+)
+library(tidyverse)
+library(vegan)
+library(tidytext)
+library(gplots)
+library(reshape2)
+library(here)
+library(tm)
+```
+
+## Set working directory and load data
+
+```{r setup}
+df <- read.csv(here("Appendix_1.csv"))
+```
+
+## Examine concepts
+
+The below example examines and plots the most common concepts in the literature sample. To select other variables, use the following in place of "`^4.`":
+
+Variable | Search String |
+--- | --- |
+Focus | `^2.` |
```

Comment:
I wrapped these `^` with \` since pandoc conversion *sometimes* uses them for special purposes (like for footnotes).